sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubAuthenticationPolicy_token.cs

```diff
@@ -0,0 +1,130 @@
+
+            //public static string GenerateAccessToken(byte[] key, string audience, string userId, TimeSpan expiresAfter)
+            //{
+            //    var expiresAt = DateTime.UtcNow.Add(expiresAfter);
+
+            //    Claim[] claims = null;
+            //    if (userId != null)
+            //    {
+            //        claims = new Claim[] { new Claim(ClaimTypes.NameIdentifier, userId) };
+            //    }
+
+            //    var securityKey = new SymmetricSecurityKey(key);
+            //    var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            //    var token = JwtTokenHandler.CreateJwtSecurityToken(
+            //        issuer: null,
+            //        audience: audience,
+            //        subject: claims == null ? null : new ClaimsIdentity(claims),
+            //        expires: expiresAt,
+            //        signingCredentials: credentials);
+
+            //    return JwtTokenHandler.WriteToken(token);
+            //}
+
+            public static string GenerateJwtBearer(
+                string audience,
+                IEnumerable<Claim> claims,
+                TimeSpan expiresAfter,
+                AzureKeyCredential key,
+                string issuer = null,
+                bool hmacSha512 = false)
+            {
+                var expiresAt = DateTime.UtcNow.Add(expiresAfter);
+
+                var subject = claims == null ? null : new ClaimsIdentity(claims);
+                SigningCredentials credentials = null;
+                if (key != null)
+                {
+                    // Refer: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/releases/tag/5.5.0
+                    // From version 5.5.0, SignatureProvider caching is turned On by default, assign KeyId to enable correct cache for same SigningKey
+                    var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(key.Key))
+                    {
+                        KeyId = KeyForCredential.GetOrCreateValue(key).Key
+                    };
+
+                    var algorithm = hmacSha512 ? SecurityAlgorithms.HmacSha512 : SecurityAlgorithms.HmacSha256;
+                    credentials = new SigningCredentials(securityKey, algorithm);
+                }
+
+                var token = JwtTokenHandler.CreateJwtSecurityToken(
+                    issuer: issuer,
+                    audience: audience,
+                    subject: subject,
+                    notBefore: null,
+                    expires: expiresAt,
+                    issuedAt: null,
+                    signingCredentials: credentials);
+                return JwtTokenHandler.WriteToken(token);
+            }
+
+            private sealed class UniqueKey
+            {
+                public string Key { get; }
+                public UniqueKey()
+                {
+                    Key = Guid.NewGuid().ToString("N");
+                }
```

Comment:
Nit - since you've got a nested class in a nested class, I'd go for brevity:

```suggestion
                public string Key { get; } = Guid.NewGuid().ToString("N");
```